### PR TITLE
[RISCV] Increase default tail duplication threshold to 6 at -O3

### PR DESCRIPTION
**Pull request by wangpc-pp**
This is just like AArch64.

Changing the threshold to 6 will increase the code size, but will
also decrease unconditional branches. CPUs with wide fetch/issue units
can benefit from it.

The value 6 may be debatable, we can set it to `SchedModel.IssueWidth`.

### CODE REVIEW

**Comment by lukel97** [comment]
Do you have any performance numbers for this? I can try and get some for the spacemit-x60.

**Comment by wangpc-pp** [comment]
> Do you have any performance numbers for this? I can try and get some for the spacemit-x60.

Thanks! I haven't run any benchmarks. I'd appreciate it if you can test it for me. :-)

**Comment by lukel97** [comment]
Sorry for the delay, I've been running SPEC CPU 2017 intrate with the changes but the results are a bit noisy, the standard deviation is about +-1%. There's been no noticeable regressions anyway (and no improvements either but I wouldn't expect any given that it's dual-issue)

**Review by preames — APPROVED** [review]
LGTM

File: llvm/test/CodeGen/RISCV/riscv-tail-dup-size.ll [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+; RUN: llc -mtriple=riscv64 -mattr=+m -tail-dup-placement-threshold=4 < %s | FileCheck %s --check-prefix=CHECK-O2
+; RUN: llc -mtriple=riscv64 -mattr=+m -tail-dup-placement-threshold=6 < %s | FileCheck %s --check-prefix=CHECK-O3
+
+@a = external dso_local local_unnamed_addr global i32
+@b = external dso_local local_unnamed_addr global i32
+@c = external dso_local local_unnamed_addr global i32
+
+declare i32 @foo(i32)
+
+define dso_local i32 @test(i32 %n) {
```
Comment (topperc) [inline llvm/test/CodeGen/RISCV/riscv-tail-dup-size.ll:15]: Do we need dso_local and local_unnamed_addr?

**Comment by wangpc-pp** [comment]
> Sorry for the delay, I've been running SPEC CPU 2017 intrate with the changes but the results are a bit noisy, the standard deviation is about +-1%. There's been no noticeable regressions anyway (and no improvements either but I wouldn't expect any given that it's dual-issue)

Thanks a lot! I just got my Spacemit K1 laptops. :-)
I was thinking that maybe we should make tail dup size a CPU-specific value so that we can set a fine-tuned value to different CPUs, just like existing `MinimumJumpTableEntries`, etc.

**Pull request merged**